crates/astria-sequencer/src/ibc/state_ext.rs

```diff
@@ -124,3 +124,278 @@ pub(crate) trait StateWriteExt: StateWrite {
 }
 
 impl<T: StateWrite> StateWriteExt for T {}
+
+#[cfg(test)]
+mod test {
+    use astria_core::sequencer::v1::{
+        asset::Id,
+        Address,
+    };
+    use cnidarium::StateDelta;
+    use ibc_types::core::channel::ChannelId;
+
+    use super::{
+        StateReadExt as _,
+        StateWriteExt as _,
+    };
+
+    #[tokio::test]
+    async fn get_ibc_sudo_address_fails_if_not_set() {
+        let storage = cnidarium::TempStorage::new().await.unwrap();
+        let snapshot = storage.latest_snapshot();
+        let state = StateDelta::new(snapshot);
+
+        // should fail if not set
+        state
+            .get_ibc_sudo_address()
+            .await
+            .expect_err("sudo address should be set");
+    }
+
+    #[tokio::test]
+    async fn put_ibc_sudo_address() {
+        let storage = cnidarium::TempStorage::new().await.unwrap();
+        let snapshot = storage.latest_snapshot();
+        let mut state = StateDelta::new(snapshot);
+
+        // can write new
+        let mut address_expected = Address::try_from_slice(&[42u8; 20]).unwrap();
+        state
+            .put_ibc_sudo_address(address_expected)
+            .expect("writing sudo address should not fail");
+        assert_eq!(
+            state
+                .get_ibc_sudo_address()
+                .await
+                .expect("a sudo address was written and must exist inside the database"),
+            address_expected,
+            "stored sudo address was not what was expected"
+        );
+
+        // can rewrite with new value
+        address_expected = Address::try_from_slice(&[41u8; 20]).unwrap();
+        state
+            .put_ibc_sudo_address(address_expected)
+            .expect("writing sudo address should not fail");
+        assert_eq!(
+            state
+                .get_ibc_sudo_address()
+                .await
+                .expect("sudo address was written and must exist inside the database"),
+            address_expected,
+            "updated sudo address was not what was expected"
+        );
+    }
+
+    #[tokio::test]
+    async fn is_ibc_relayer_ok_if_not_set() {
+        let storage = cnidarium::TempStorage::new().await.unwrap();
+        let snapshot = storage.latest_snapshot();
+        let state = StateDelta::new(snapshot);
+
+        // unset address returns false
+        let address_expected = Address::try_from_slice(&[42u8; 20]).unwrap();
```

Comment:
```suggestion
        let address = Address::try_from_slice(&[42u8; 20]).unwrap();
```

Reply:
Agree, I suggested the `expected` nomenclature in a different PR, but here in fact no tests on an "expected" address are done - `address` is an input to a method call and it is that method call that is being verified.